components/brave_wallet/common/eth_sign_typed_data_helper.cc

```diff
@@ -0,0 +1,327 @@
+      if (!encoded_item)
+        return absl::nullopt;
+      array_result.insert(array_result.end(), encoded_item->begin(),
+                          encoded_item->end());
+    }
+    auto array_hash = KeccakHash(array_result);
+    result.insert(result.end(), array_hash.begin(), array_hash.end());
+  } else if (type == "string") {
+    const std::string* value_str = value.GetIfString();
+    if (!value_str)
+      return absl::nullopt;
+    const std::string encoded_value = KeccakHash(*value_str, false);
+    const std::vector<uint8_t> encoded_value_bytes(encoded_value.begin(),
+                                                   encoded_value.end());
+    result.insert(result.end(), encoded_value_bytes.begin(),
+                  encoded_value_bytes.end());
+  } else if (type == "bytes") {
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> bytes;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &bytes));
+    const std::vector<uint8_t> encoded_value = KeccakHash(bytes);
+    result.insert(result.end(), encoded_value.begin(), encoded_value.end());
+  } else if (type == "bool") {
+    absl::optional<bool> value_bool = value.GetIfBool();
+    if (!value_bool)
+      return absl::nullopt;
+    uint256_t encoded_value = (uint256_t)*value_bool;
+    for (int i = 256 - 8; i >= 0; i -= 8) {
+      result.push_back((encoded_value >> i) & 0xFF);
+    }
+  } else if (type == "address") {
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> address;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &address));
+    DCHECK_EQ(address.size(), 20u);
+    for (size_t i = 0; i < 256 - 160; i += 8)
+      result.push_back(0);
+    result.insert(result.end(), address.begin(), address.end());
+
+  } else if (base::StartsWith(type, "bytes", base::CompareCase::SENSITIVE)) {
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 5, &type_check) || type_check > 32)
+      return absl::nullopt;
+    const std::string* value_str = value.GetIfString();
+    if (!value_str || !IsValidHexString(*value_str))
+      return absl::nullopt;
+    std::vector<uint8_t> bytes;
+    CHECK(base::HexStringToBytes(value_str->data() + 2, &bytes));
+    if (bytes.size() > 32)
+      return absl::nullopt;
+    result.insert(result.end(), bytes.begin(), bytes.end());
+    for (size_t i = 0; i < 32u - bytes.size(); ++i) {
+      result.push_back(0);
+    }
+  } else if (base::StartsWith(type, "uint", base::CompareCase::SENSITIVE)) {
+    // uint8 to uint256 in steps of 8
+    unsigned type_check;
+    if (!base::StringToUint(type.data() + 4, &type_check) || type_check % 8 ||
+        type_check > 256)
+      return absl::nullopt;
+
+    absl::optional<int> value_int = value.GetIfInt();
```

Comment:
Since this is from JSON, I think we should use 
```
// ES6 section 20.1.2.6 Number.MAX_SAFE_INTEGER
constexpr uint64_t kMaxSafeIntegerUint64 = 9007199254740991;  // 2^53-1
constexpr double kMaxSafeInteger = static_cast<double>(kMaxSafeIntegerUint64);

``` 

instead so we can get bigger numbers that would be valid in JSON.

Reply:
actually even better is using DBL_MAX or which I think is the same as numeric limits for double.
```
The biggest/largest integer that can be stored in a double without losing precision is the same as the largest possible value of a double. That is, DBL_MAX or approximately 1.8 × 10308 (if your double is an IEEE 754 64-bit double). It's an integer.
```

Reply:
fixed in 6bbb1c4516cb09859c699807dc35b5fd17d06b54